src/apis/localization.js

```diff
@@ -139,3 +139,109 @@ export const getLocaleData = ({ locale }) => {
 export const setLanguage = ({ lang }) => {
     return setProperty("Language", cockpit.variant("s", lang));
 };
+
+const normalizeLanguage = (lang) => {
+    const parts = lang.split(".")[0].split("_");
+    return parts.length > 1 ? parts[1].toLowerCase() : parts[0];
+};
+
+/**
+ * Sets the system keyboard layout and variant.
+ * @param {string} keyboardValue - A string in the format "layoutId:variantId"
+ */
+export const setKeyboardLayout = async (keyboardValue) => {
+    const [layoutId, variantId] = keyboardValue.split(":");
+
+    try {
+        const command = ["localectl", "set-x11-keymap", layoutId];
+        if (variantId && variantId !== "default") {
+            command.push(variantId);
+        }
+
+        await cockpit.spawn(command);
+    } catch (error) {
+        console.error(`Failed to set keyboard layout (${layoutId}, ${variantId}):`, error);
+        throw error;
+    }
+};
+
+/**
+ * @param {string} lang         Language id
+ * @returns {Promise}           Resolves a list of keyboard layouts for the specified language
+ *                              in format {description:, id:}
+ */
+export const getKeyboardLayoutsForLanguage = async (lang) => {
+    const normalizedLang = normalizeLanguage(lang);
+    const filePath = "/usr/share/X11/xkb/rules/evdev.lst";
```

Comment:
Parsing this file sounds like not the most robust solution. I dont object it, but maybe we could think of something more stable? 

This info could be retrieved I think from the python bindings of xdb, python-xkbcommon

So we could have a python script that generates this, converts it into a JSON file, which we can easily consume from Javascript. 

@jkonecny12 what do you think? Do we want this solution, as long as it's tested and wait for Gnome to resolve this by implementing the localed APIs for Workstation?